### PR TITLE
test(py): add pytest-bdd feature tests for csi

### PR DESCRIPTION
**Pull request by cjones1024**
- roughly translated from test/grpc/test_csi.js

- updated README.md to add command for generating python files
  from protobuf definitions in csi/proto/csi.proto
  to setup instructions

- also added a non-bdd test which tests both stage/unstage
  and stage/publish/unpublish/unstage operations for each
  combination of: share_type + fs_type + acccess_mode + ro/rw

- resolves CAS-967

### CODE REVIEW

**Review by gila — DISMISSED** [review]
` AttributeError: module 'os' has no attribute 'env' ` this all fails for me with that error.

File: test/python/test_bdd_csi.py [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+    "Staging the same volumes with a different staging_target_path."
+
+
+@scenario("features/csi.feature", "unstaging a single writer volume")
+def test_unstaging_a_single_writer_volume():
+    "Unstaging a single writer volume."
+
+
+@pytest.fixture
+def published_nexuses(mayastor_instance, mayastor_nexuses):
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply test/python/test_bdd_csi.py:327]: oh I see how it works now :+1:

**Review by tiagolobocastro — APPROVED** [review]

File: test/python/test_bdd_csi.py [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+
+@pytest.fixture(scope="module")
+def start_csi_plugin():
+    def monitor(proc, result):
+        stdout, stderr = proc.communicate()
+        result["stdout"] = stdout.decode()
+        result["stderr"] = stderr.decode()
+        result["status"] = proc.returncode
+
+    proc = subprocess.Popen(
```
[the remark this answers is not in the document]
Comment (cjones1024) [reply test/python/test_bdd_csi.py:96]: Ultimately it should be run in a container. However, it does require running in privileged mode, and other configuration changes are required. The upshot is that in order to do this we need, at the very least, a mechanism for specifying a per-test-suite docker-compose configuration which we currently we don't have. For now, we start and stop the plugin manually on the test host, but this is only a temporary measure.

File: test/python/test_bdd_csi.py [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+
+def get_uuid(n):
+    return "11111111-0000-0000-0000-%.12d" % (n)
+
+
+def share_type(protocol):
+    import mayastor_pb2
+
+    TYPES = {
+        "nbd": mayastor_pb2.ShareProtocolNexus.NEXUS_NBD,
```
[the remark this answers is not in the document]
Comment (cjones1024) [reply test/python/test_bdd_csi.py:53]: I'm not sure what the official position is but I suspect not. However, this is still defined in out protobuf definition for the gRPC api. In theory we still should be able to test with nbd, even if it is only to verify that it is not supported.

File: test/python/test_bdd_csi.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply test/python/test_bdd_csi.py:96]: :+1:

**Comment by cjones1024** [comment]
bors try

**Comment by cjones1024** [comment]
bors try

**Review by paulyoong — APPROVED** [review]

**Comment by cjones1024** [comment]
bors merge

**Pull request merged**